source/modulo_components/include/modulo_components/ComponentInterface.hpp

```diff
@@ -399,11 +399,22 @@ inline void ComponentInterface<NodeT>::add_parameter(
       rcl_interfaces::msg::ParameterDescriptor descriptor;
       descriptor.description = description;
       descriptor.read_only = read_only;
-      NodeT::declare_parameter(parameter->get_name(), ros_param.get_parameter_value(), descriptor);
+      if (parameter->is_empty()) {
+        descriptor.dynamic_typing = true;
+        NodeT::declare_parameter(
+            parameter->get_name(), utilities::get_ros_parameter_type(parameter->get_parameter_type()));
```

Comment:
Shouldn't `descriptor` be third argument there?

Reply:
Oh that might solve my problem!

Reply:
It doesn't...you cannot specify the type as second argument and allow dynamic typing at the same time